src/com/google/javascript/jscomp/parsing/parser/Scanner.java

```diff
@@ -869,52 +870,18 @@ private static String processUnicodeEscapes(String value) {
     return value;
   }
 
-  @SuppressWarnings("ShortCircuitBoolean") // Intentional to minimize branches in this code
+  /**
+   * Interface from UnicodeRegex. Includes old optimizations.
+   */
   private static boolean isIdentifierStart(char ch) {
-    // Most code is written in pure ASCII, so create a fast path here.
-    if (ch <= 127) {
-      // Intentionally avoiding short circuiting behavior of "||" and "&&".
-      // This minimizes branches in this code which minimizes branch prediction misses.
-      return ((ch >= 'A' & ch <= 'Z') | (ch >= 'a' & ch <= 'z') | (ch == '_' | ch == '$'));
-    }
-
-    // Handle non-ASCII characters.
-    // TODO(tjgq): This should include all characters with the ID_Start property.
-    if (Character.isLetter(ch)) {
```

Comment:
Do you feel this is insufficient for the native Java builds?   Where do you believe you need the JavaScript version?

Reply:
@concavelenz I believe the reason I removed `Character.isLetter` was because the GWT build would fail, as J2CL / Guava implementations did not exist or would not compile to JS. See: google/j2cl#103

Apologies for the lack of specificity, this issue is about a month old and is not super fresh in my mind. Thank you for your reviews, John.